pkg/proxy/topology_test.go

```diff
@@ -476,3 +476,138 @@ func TestFilterTopologyEndpoint(t *testing.T) {
 		})
 	}
 }
+
+func TestFilterLocalEndpoint(t *testing.T) {
+	type endpoint struct {
+		Endpoint string
+		IsLocal  bool
+	}
+	all := v1.ServiceInternalTrafficPolicyAll
+	local := v1.ServiceInternalTrafficPolicyLocal
+	preferLocal := v1.ServiceInternalTrafficPolicyPreferLocal
+
+	testCases := []struct {
+		Name                  string
+		internalTrafficPolicy *v1.ServiceInternalTrafficPolicyType
+		endpoints             []endpoint
+		expected              []endpoint
+	}{
+		{
+			// Case[0]: no internalTrafficPolicy set with empty
```

Comment:
really minor nit: we probably don't need these comments since we have the same description in the test names.